packages/edit-site/src/components/header/index.js

```diff
@@ -79,6 +87,11 @@ export default function Header( {
 	const displayBlockToolbar =
 		! isLargeViewport || deviceType !== 'Desktop' || hasFixedToolbar;
 
+	let { title } = getTemplateInfo( template );
```

Comment:
+1
As far as I can see, pretty much all the changes to this file can live in `DocumentActions`, which would become a component without props. 🙂

Reply:
The reason I wrote it like this is so that it could be reused in `edit-post` eventually. Otherwise it is hardcoded to _just_ the template, active template part, or site editor. The refactor makes it basically standalone so that you could pass through an arbitrary _thing_ to show in the header, and the component just puts the info into the right spot. 

I personally think that it's a better separation of concerns

Reply:
Two things gave me pause:

1. Until we create shared header abstraction, prop drilling adds an extra layer of complexity. 
2. We may have a potentially premature abstraction

[It does, however, sound like we're certain about implementing the same component in edit-post and edit-site, so reuse makes sense](https://github.com/WordPress/gutenberg/issues/20877#issuecomment-707032028). The prop drilling also looks manageable since it's only 1 level deep. This implementation looks good to me, but I'm curious about other opinions as well.

Reply:
Got it, didn't think of reusability!

If that's the intention, then I guess we could actually move `DocumentActions` in a shared package? 🤔 
I'm not sure if the `components` package would be a good fit, though... maybe `editor`?

Reply:
Right, I'm not sure about that. The reason I added the abstraction now is just so that it's easier to manage templates vs template parts (e.g. handling the "different types of entities" part of the abstraction). to me, i don't think we need to move it into a different package until we're ready. but it does need to handle different types of entities right now

Reply:
👍 Sounds good to me

Reply:
I didn't consider potential reusability with `edit-post` when I was writing the comment. With that in mind this makes sense. 👍